eZ/Publish/Core/Persistence/Legacy/Content/Mapper.php

```diff
@@ -361,7 +361,8 @@ public function extractVersionInfoListFromRows(array $rows, array $nameRows)
                 $versionInfo->creationDate = (int)$row['ezcontentobject_version_created'];
                 $versionInfo->modificationDate = (int)$row['ezcontentobject_version_modified'];
                 $versionInfo->status = (int)$row['ezcontentobject_version_status'];
-                $versionInfo->names = $nameData[$versionId];
+                // Versions with VersionInfo::STATUS_INTERNAL_DRAFT might not have a row in ezcontentobject_name
+                $versionInfo->names = isset($nameData[$versionId]) ? $nameData[$versionId] : null;
```

Comment:
```suggestion
                $versionInfo->names = $nameData[$versionId]) ?? [];
```

It will be safer to assign empty array. See https://github.com/ezsystems/ezplatform-kernel/blob/b79252a6040048e826e085fbed099b942c693a39/eZ/Publish/Core/Repository/Values/Content/VersionInfo.php#L33-L34